MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -3038,4 +3048,81 @@ - (BOOL)canEditEventWithId:(NSString*)eventId
     return [self canPerformActionOnEvent:event] && [messageType isEqualToString:kMXMessageTypeText] && [event.eventId isEqualToString:self.mxSession.myUser.userId];
 }
 
+- (void)registerEventEditsListener
+{
+    MXWeakify(self);
+    eventEditsListener = [self.mxSession.aggregations listenToEditsUpdateInRoom:self.roomId block:^(MXEvent * _Nonnull replaceEvent) {
+        MXStrongifyAndReturnIfNil(self);
+        
+        NSString *editedEventId = replaceEvent.relatesTo.eventId;
+        
+        dispatch_async(MXKRoomDataSource.processingQueue, ^{
+            
+            // Check whether a message contains the redacted event
+            id<MXKRoomBubbleCellDataStoring> bubbleData = [self cellDataOfEventWithEventId:editedEventId];
+            if (bubbleData)
+            {
+                BOOL hasChanged = [self updateCellData:bubbleData forEditionWithReplaceEvent:replaceEvent andEventId:editedEventId];
+                
+                if (hasChanged)
+                {
+                    // Update the delegate on main thread
+                    dispatch_async(dispatch_get_main_queue(), ^{
+                        
+                        if (self.delegate)
+                        {
+                            [self.delegate dataSource:self didCellChange:nil];
+                        }
+                        
+                    });
+                }
+            }
+        });
+    }];
+}
+
+- (void)unregisterEventEditsListener
+{
+    if (receiptsListener)
+    {
+        [self.mxSession.aggregations removeListener:receiptsListener];
+        receiptsListener = nil;
+    }
+}
+
+- (BOOL)updateCellData:(id<MXKRoomBubbleCellDataStoring>)bubbleCellData forEditionWithReplaceEvent:(MXEvent*)replaceEvent andEventId:(NSString*)eventId
+{
+    BOOL hasChanged = NO;
+    
+    @synchronized (bubbleCellData)
+    {
+        // Retrieve the original event to edit it
+        NSArray *events = bubbleCellData.events;
+        MXEvent *editedEvent = nil;
+
+        // TODO: Handle non live timeline in memory store (permalink)
```

Comment:
I think you do the TODO few lines below (https://github.com/matrix-org/matrix-ios-kit/pull/562/files#diff-f65a2dab99c08c685dede0cb12eec13fR3111)